extensions/script-libraries/oauth2/oauth2.livecodescript

```diff
@@ -0,0 +1,232 @@
+﻿script "com.livecode.script-library.oauth2"
+/*
+Copyright (C) 2016 LiveCode Ltd.
+
+This file is part of LiveCode.
+
+LiveCode is free software; you can redistribute it and/or modify it under
+the terms of the GNU General Public License v3 as published by the Free
+Software Foundation.
+
+LiveCode is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or
+FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+for more details.
+
+You should have received a copy of the GNU General Public License
+along with LiveCode.  If not see <http://www.gnu.org/licenses/>. 
+*/
+
+/**
+
+Name: OAuth2
+
+Summary:
+Present an authorization dialog for any web service that supports OAuth2 Authorization Code Flow 
+
+Requires:
+
+- mergJSON
+
+*/
+
+on revLoadLibrary
+   if the target is not me then
+      pass revLoadLibrary
+   end if
+   
+   insert the script of me into back
+   
+   if the environment contains "development" then
+      set the _ideoverride of me to true
+      revSBAddAvailableLibrary the short name of me, "OAuth2"
+      revSBAddDependencyForInclusion "scriptLibraries", "OAuth2", "externals", "mergJSON"
+   end if
+end revLoadLibrary
+
+on revUnloadLibrary
+   if the target is not me then
+      pass revUnloadLibrary
+   end if
+   
+   remove the script of me from back
+end revUnloadLibrary
+
+constant kRedirectURL = "http://localhost"
+local sPort = 8080
+
+local sUniqueRef
+
+/**
+
+Present an authorization dialog for any web service that supports OAuth2 Authorization Code Flow
+
+Description:
+In order to handle the redirect the library accepts socket connections on localhost on a
+configurable port. The redirect uri configured when setting up your application with the
+web service should be `http://localhost:<port>` where `<port>` is the port that can be
+configured with `OAuth2Set "port",<port>`. The default port is 8080.
```

Comment:
@peter-b with regard to multiple dialogs perhaps the simplest is if there is already a dialog open then `return "dialog open" for error` or something...